MetaMask/metamask-extension
# Align network client tests with core repo

## Explanation



When the network client tests in this repo were copied over to the core repo, some alterations were made. This commit copies them back to this repo so that the two test suites are easier to compare to each other. Changes include:

* Extracting `testsFor*` functions (functions that bundle similar tests together) into separate files
* Reordering the tests defined in `testsForRpcMethodsThatCheckForBlockHashInResponse`
* Reordering the set of RPC methods listed in `shared-tests.js`
* Reordering the type definitions defined in `shared-tests.js` to be closer to the functions that use them
* Updating the tests defined in `testsForRpcMethodSupportingBlockParam` so that when they make requests they pass params, even if it's just an empty array

For more context, see: https://github.com/MetaMask/core/pull/1087

## Manual Testing Steps



This commit makes no functional changes — it only makes changes to tests. All of the tests should pass, but if you want to manually verify, run:

```
yarn jest app/scripts/controllers/network/{createInfuraClient.test.js,createJsonRpcClient.test.js}
```

## Pre-merge author checklist


## Pre-merge reviewer checklist


If further QA is required (e.g. new feature, complex testing steps, large refactor), add the `Extension QA Board` label.

In this case, a QA Engineer approval will be be required.
